core/validatorInfo/validatorInfoUtils.go

```diff
@@ -0,0 +1,19 @@
+package validatorInfo
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+)
+
+// IsLeavingEligible returns true if the validator was eligible in the epoch but has done an unstake.
+func IsLeavingEligible(valInfo *state.ValidatorInfo) bool {
+	if valInfo == nil {
```

Comment:
Maybe this could be renamed to WasEligibleInCurrentEpoch and would return true even if validatorInfo.List == string(core.eligible)

Reply:
What happens with the validators who are jailed but kept in the system as eligible?